_integrator-guide/jsonata-mapper-dates-times.md

```diff
@@ -0,0 +1,219 @@
+
+*   6st of December 2017 (DD.MM.YYYY) = `"2017-12-05T23:00:00.000Z"`. Converted to UTC and no DST due to the time of the year.
+*   12th of June 2017 (MM.DD.YYYY) = `"2017-06-11T22:00:00.000Z"`. Converted to UTC and + DST.
+
+In above case, it was assumed the German Time (Central Europen Time zone). It
+will get complicated if it is not, therefore, one should prevent this inconsistency
+in advance by using known [ISO 8601 standard](https://en.wikipedia.org/wiki/ISO_8601)
+or String + Format method:
+
+```js
+// String + format
+$moment('06.12.2017','DD.MM.YYYY')
+// or ISO 8601
+$moment('2017-12-06')
+```
+
+Both implementations will return the same value:
+
+```
+"2017-12-05T23:00:00.000Z"
+```
+
+## Formatting the outcome
+
+`$moment()` returns the date and time in a standard ISO 8601 format but it can
+be modified by using .format() parameter. When used in the initial empty state:
+
+```js
+$moment('2017-12-06').format() & "\n" &
+$moment('2017-06-12').format()
+```
+
+Would return:
+
+```
+2017-12-06T00:00:00+01:00
+2017-06-12T00:00:00+02:00
+```
+
+These are the ISO 8601 formatted date and the time values offset from UTC at
+each particular moment (German Time). It is also possible to output using many
+other formats. For example:
+
+```js
+$moment('2012-12-01').format('MM/DD/YYYY')
+```
+
+This expression returns the input ISO 8601 value in a formatted US date format.
+
+```
+12/01/2012
+```
+
+Let us take another example:
+
+```js
+$moment('2012-12-01T22:32:16').format("dddd, MMMM Do YYYY, h:mm:ss a")
+```
+
+This outcome of this expression would be:
+
+```
+Saturday, December 1st 2012, 10:32:16 pm
+```
+
+More details about possible formatting is available from the
+[Moment.js documentation pages on formatting](https://momentjs.com/docs/#/displaying/format/).
+
+## Date and time operations
+
+Any given date one can be changed using manipulate functions. It is possible to
+add and subtract years, months, weeks, days and etc. For example to add 1 year,
+3 months and 15 days from any particular date:
+
+```js
+$moment('2012-12-01').add(1, 'year').add(3,'months').add(15,'days').format('LL')
+```
+
+The outcome would be:
+
+```
+March 16, 2014
+```
+
+It is also possible to use add and subtract together in one expression like this:
+
+```js
+$moment('2012-12-01').add(3, 'year').subtract(25,'days').format('LL')
+```
+
+The answer would be:
+
+```
+November 6, 2015
+```
+
+**Warning: Date and time operations are not linear in nature.** For example,
+daylight saving time (DST) can cause the day to be 23 or 25 hours long.
+Similarly, leap years need to be taken into account during the calculations.
+Therefore, add/subtract dates and time separately. More information [here](http://momentjs.com/guides/#/lib-concepts/date-time-math/).
+
+[Moment.js documentation](https://momentjs.com/docs/#/manipulating/) includes
+considerably more possibilities to manipulate the date and time.
+
+## Warning: Parsing Inconsistency pitfall
+
```

Comment:
@jhorbulyk where is the troubleshooting part in this? It is still informative document about the `date` and `time` using `$moment( )`. Note that no errors are presented to point the reader that if you see this do that.

Reply:
>  Note that no errors are presented to point the reader that if you see this do that.

That is technically true.  However, I feel as though the magnitude of importance of this section is less that the magnitude of importance of the other sections.  It feels as though it could belong in some sort of "tip box".  However, that statement is pretty subjective.